<a href="https://colab.research.google.com/github/Sun-cs-Sol/DesafioDados1-7/blob/main/DesafioDados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

Base de gastos dos deputados: Cota para Exercício da Atividade Parlamentar dos Senadores (CEAPS).

Objetivo: tratar a base e poder tirar insghts reais.


Informações sobre a base de dados:

- Coluna -> MES: fala sobre o número correspondente ao mês do gasto.
- Coluna -> SENADOR: Fala sobre o nome do senador que realizou o gasto.
- Coluna -> TIPO_DESPESA: Categoria na qual a despesa está alocada.
- Coluna ->  CNPJ_CPF: Cadastro no receberdor do pagamento
- Coluna ->  FORNECEDOR: Nome do recebedor do pagamento
- Coluna ->  DATA: data da realização do pagamento
- Coluna ->  Detalhamento: descrição do pagamento
- Coluna ->  VALOR_REEMBOLSADO: Valor gasto no pagamento pelo governo






In [5]:
df = pd.read_csv('/content/despesa_ceaps_2022.csv', header=None)
df.drop(1, inplace=True)

df.columns = ["ANO", "MES", "SENADOR", "TIPO_DESPESA", "CNPJ_CPF", "FORNECEDOR","DOCUMENTO", "DATA", "DETALHAMENTO", "VALOR_REEMBOLSADO", "COD_DOCUMENTO"]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16806 entries, 0 to 16806
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   ANO                16806 non-null  object
 1   MES                16806 non-null  object
 2   SENADOR            16805 non-null  object
 3   TIPO_DESPESA       16805 non-null  object
 4   CNPJ_CPF           16805 non-null  object
 5   FORNECEDOR         16805 non-null  object
 6   DOCUMENTO          15871 non-null  object
 7   DATA               16805 non-null  object
 8   DETALHAMENTO       9062 non-null   object
 9   VALOR_REEMBOLSADO  16805 non-null  object
 10  COD_DOCUMENTO      16805 non-null  object
dtypes: object(11)
memory usage: 1.5+ MB


Olhando as quantidade de linhas não nulas pro coluna pode-se observar que as colunas 6 (DOCUMENTO) e 8 (DETALHAMENTO) são as unicas que contém valores nulos intercalados nos dados reais. As colunas não estou agregando informações fundamentais dentro da tabela então criar uma copia dela e vamos excluir elas para manipular os dados mais relevantes.


In [7]:
df1 = df.copy()
df1 = df1.drop(columns=["DOCUMENTO", "DETALHAMENTO"])
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16806 entries, 0 to 16806
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   ANO                16806 non-null  object
 1   MES                16806 non-null  object
 2   SENADOR            16805 non-null  object
 3   TIPO_DESPESA       16805 non-null  object
 4   CNPJ_CPF           16805 non-null  object
 5   FORNECEDOR         16805 non-null  object
 6   DATA               16805 non-null  object
 7   VALOR_REEMBOLSADO  16805 non-null  object
 8   COD_DOCUMENTO      16805 non-null  object
dtypes: object(9)
memory usage: 1.3+ MB


Agora vamos padronizar os valores da coluna VALOR_REEMBOLSADO

In [9]:
df["VALOR_REEMBOLSADO"] = df["VALOR_REEMBOLSADO"].astype(str).str.replace(",", ".").astype(float)
print(df["VALOR_REEMBOLSADO"])

0            NaN
2        6000.00
3        1500.00
4        6000.00
5        1000.00
          ...   
16802    2893.04
16803    1180.19
16804    2671.90
16805    1334.31
16806    2250.72
Name: VALOR_REEMBOLSADO, Length: 16806, dtype: float64


Base limpa! agora podemos começar a manipular dos dados para extrair insights.

Vamos descobrir de quantos deputados estamos falando.

In [ ]:
senadores = df["SENADOR"].unique()
valor = df["SENADOR"].count()
print(senadores)
print(f"\nQuantidade de senadores:   {len(senadores)}")
print(f"Valore médio por senador: {valor/len(senadores):.2f} pedidos")

[nan 'ACIR GURGACZ' 'ALESSANDRO VIEIRA' 'ALEXANDRE SILVEIRA' 'ALVARO DIAS'
 'ANGELO CORONEL' 'ANTONIO ANASTASIA' 'CARLOS FÁVARO' 'CARLOS PORTINHO'
 'CARLOS VIANA' 'CHICO RODRIGUES' 'CHIQUINHO FEITOSA' 'CID GOMES'
 'CONFÚCIO MOURA' 'DANIELLA RIBEIRO' 'DÁRIO BERGER' 'DAVI ALCOLUMBRE'
 'DRA. EUDÓCIA' 'EDUARDO BRAGA' 'EDUARDO GIRÃO' 'EDUARDO GOMES'
 'EDUARDO VELLOSO' 'ELIANE NOGUEIRA' 'ELIZIANE GAMA' 'ELMANO FÉRRER'
 'ESPERIDIÃO AMIN' 'FABIANO CONTARATO' 'FABIO GARCIA'
 'FERNANDO BEZERRA COELHO' 'FERNANDO COLLOR' 'FERNANDO DUEIRE'
 'FLÁVIO ARNS' 'FLÁVIO BOLSONARO' 'GIORDANO' 'GUARACY SILVEIRA'
 'HUMBERTO COSTA' 'IRAJÁ' 'IVETE DA SILVEIRA' 'IZALCI LUCAS'
 'JADER BARBALHO' 'JAQUES WAGNER' 'JARBAS VASCONCELOS' 'JAYME CAMPOS'
 'JEAN PAUL PRATES' 'JORGINHO MELLO' 'JOSÉ ANÍBAL' 'JOSÉ SERRA'
 'JULIO VENTURA' 'KÁTIA ABREU' 'LASIER MARTINS' 'LEILA BARROS'
 'LUCAS BARRETO' 'LUIS CARLOS HEINZE' 'LUIZ CARLOS DO CARMO'
 'LUIZ PASTORE' 'MAILZA GOMES' 'MARA GABRILLI' 'MARCELO CASTRO'
 'MARCIO BITTAR' 'MA

Agora sabemos que as 16805 requisições de reembolso foram feitas por 98 senadores e a média de requisições por senador é de 171 aproximadamente.